plugins/inputs/webhooks/webhooks.go

```diff
@@ -0,0 +1,81 @@
+package webhooks
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	_ "github.com/influxdata/telegraf/plugins/inputs/webhooks/webhooks_all"
+	"github.com/influxdata/telegraf/plugins/inputs/webhooks/webhooks_models"
+)
+
+func init() {
+	inputs.Add("webhooks", func() telegraf.Input { return NewWebhooks() })
+}
+
+type Webhooks struct {
+	ServiceAddress string
+
+	Webhook []WebhookConfig
+}
+
+type WebhookConfig struct {
```

Comment:
Number 2 is true, you would need to maintain a list of the available webhook plugins.

Not sure I understand what you mean by Number 1, you should just be able to check if `Github` is not nil, and start it if so.